kube-client/src/client/mod.rs

```diff
@@ -238,15 +237,18 @@ impl Client {
         Ok(text)
     }
 
-    /// Perform a raw HTTP request against the API and get back the response
-    /// as a stream of bytes
-    pub async fn request_text_stream(
-        &self,
-        request: Request<Vec<u8>>,
-    ) -> Result<impl Stream<Item = Result<Bytes>>> {
+    /// Perform a raw HTTP request against the API, get back the response
+    /// and return it as an object that implements [`AsyncBufRead`] and thus
+    /// [`AsyncBufReadExt`]. Users can call `.lines()` on it to get a newline
+    /// buffered stream of logs.
+    pub async fn request_text_stream(&self, request: Request<Vec<u8>>) -> Result<impl AsyncBufRead> {
         let res = self.send(request.map(Body::from)).await?;
-        // trace!("Status = {:?} for {}", res.status(), res.url());
-        Ok(res.into_body().map_err(Error::HyperError))
+        // Map the error as we want to convert this into an `AsyncBufReader` using
+        // `into_async_read` which specifies `std::io::Error` as the stream's error type.
+        let body = res
+            .into_body()
+            .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e));
```

Comment:
Appreciate this comment. This felt like an awkward re-nesting of an already good error, but it does seem to be required upstream :(